src/main/java/aquality/selenium/elements/ElementFactory.java

```diff
@@ -60,9 +62,20 @@ protected Map<Class<? extends aquality.selenium.core.elements.interfaces.IElemen
      */
     @Override
     protected By generateXpathLocator(By multipleElementsLocator, WebElement webElement, int elementIndex) {
```

Comment:
It looks weird that we are trying to generate css locator inside the generate**Xpath**Locator.
In you case of shadow elements  - wouldn't it be better to have separate method like generateCssLocator and use it for shadow elements?
Below I see we are using separate method for work with shadow elements: findElementInShadowRoot
Can't we always run generateCssLocator inside of it?

Reply:
We don't need to generate locator when we do findElementInShadowRoot, it is needed for finding multiple elements from the shadowRoot-parent. We are not differentiating whether element is inside-of-shadow or not, it differs only by search context.
I agree that naming confuses, but it comes from the core library. Need to do renaming in the core ElementFactory

Reply:
fixed